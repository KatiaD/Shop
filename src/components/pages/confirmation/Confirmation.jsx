import React from 'react';
import PropTypes from 'prop-types';
import Master from 'composables';
import { Link } from 'react-router-dom';
import { CartItem } from 'components/widgets';
import { CartTable } from '../cart/Cart.styled';

const displayName = 'Confirmation';

const propTypes = {
  myProducts: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired,
  total: PropTypes.number.isRequired,
  quantity: PropTypes.objectOf(PropTypes.number),
  values: PropTypes.objectOf(PropTypes.object).isRequired,
};

const defaultProps = {
  quantity: {},
};

function Confirmation({
  myProducts, total, quantity, values,
}) {
  return (
    <Master title="Confirmation">
      {total ? (
        <CartTable>
          <thead>
            <tr>
              <th>Image</th>
              <th>Product name</th>
              <th>Price</th>
              <th>Quantity</th>
              <th>total</th>
            </tr>
          </thead>
          <tbody>
            {myProducts.map(product => (
              <CartItem key={product.id} {...product} controls={false} quantity={quantity[product.id]} />
            ))}
          </tbody>
        </CartTable>
      ) : (
        'Cart is empty'
      )}
      {total ? `TOTAL: ${total}` : <Link to="/">Back to catalog</Link>}
      <div data-name="user-info-name">
        {values.name}
      </div>
      <div data-name="user-info-email">
        {values.email}
      </div>
      <div data-name="user-info-age">
        {values.age}
      </div>
    </Master>
  );
}

Confirmation.displayName = displayName;
Confirmation.propTypes = propTypes;
Confirmation.defaultProps = defaultProps;

export default Confirmation;
