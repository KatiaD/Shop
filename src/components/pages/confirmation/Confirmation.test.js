import React from 'react';
import { shallow } from 'enzyme';

import Confirmation from './Confirmation';

const testProps = {
  myProducts: [],
  total: 500,
  values: {},
  quantity: {},
};

describe('Given a ConfirmationPage component', () => {
  describe('when the component is rendered', () => {
    let component;

    beforeAll(() => {
      component = shallow(<Confirmation {...testProps} />);
    });

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
