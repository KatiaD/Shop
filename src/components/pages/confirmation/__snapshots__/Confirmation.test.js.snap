// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ConfirmationPage component when the component is rendered should match the snapshot 1`] = `
<Master
  title="Confirmation"
>
  <Cartstyled__CartTable>
    <thead>
      <tr>
        <th>
          Image
        </th>
        <th>
          Product name
        </th>
        <th>
          Price
        </th>
        <th>
          Quantity
        </th>
        <th>
          total
        </th>
      </tr>
    </thead>
    <tbody />
  </Cartstyled__CartTable>
  TOTAL: 500
  <div
    data-name="user-info-name"
  />
  <div
    data-name="user-info-email"
  />
  <div
    data-name="user-info-age"
  />
</Master>
`;
